irohad/torii/processor/impl/transaction_processor_impl.cpp

```diff
@@ -121,19 +128,26 @@ namespace iroha {
       });
     }
 
-    void TransactionProcessorImpl::batchHandle(
+    bool TransactionProcessorImpl::batchHandle(
         std::shared_ptr<shared_model::interface::TransactionBatch>
             transaction_batch) const {
       log_->info("handle batch");
+      bool batch_passed_for_further_processing(true);
       if (transaction_batch->hasAllSignatures()
           and not mst_processor_->batchInStorage(transaction_batch)) {
         log_->info("propagating batch to PCS");
         this->publishEnoughSignaturesStatus(transaction_batch->transactions());
-        pcs_->propagate_batch(transaction_batch);
+        log_->critical("BATCH {}", transaction_batch);
+        if (not(batch_passed_for_further_processing =
+                    pcs_->propagate_batch(transaction_batch))) {
+          log_->info("PCS was unable to serve the batch {}",
```

Comment:
Is it required to log here? Maybe the log in CommandServiceTransport is enough. In this case the code here can be simplified to the following, without using the addidional `batch_passed_for_further_processing` variable.
```C++
return pcs_->propagate_batch(transaction_batch);
```